caffe2/sgd/adadelta_op_gpu.cu

```diff
@@ -0,0 +1,168 @@
+#include "adadelta_op.h"
+#include "caffe2/core/common_gpu.h"
+#include "caffe2/core/context_gpu.h"
+#include "caffe2/utils/mixed_utils.h"
+
+namespace caffe2 {
+
+__global__ void AdadeltaUpdate(
+    int N,
+    const float* w,
+    const float* g,
+    const float* h,
+    const float* d,
+    float* nw,
+    float* nh,
+    float* nd,
+    float epsilon,
+    float decay,
+    const float* lr) {
+  CUDA_1D_KERNEL_LOOP(i, N) {
+    float gi = g[i];
+    float di = d[i];
+    float hi = nh[i] = decay * h[i] + (1.0f - decay) * gi * gi;
+    float ng = (std::sqrt(di + epsilon) / std::sqrt(hi + epsilon)) * gi;
```

Comment:
Please do not use std:: functions in kernel. Please use sqrtf and rsqrtf here.
I'd suggest
const float ng =  sqrtf(__ldg(d + i) + epsilon) * rsqrtf(__ldg(nh + i) + epsilon) * __ldg(g + i);